components/admin/AdminFilter.tsx

```diff
@@ -0,0 +1,155 @@
+import { FirebaseTablesEnum, StatusEnum } from "@/lib/enums";
+import { cn } from "@/lib/utils";
+import { Button } from "../ui/button";
+import { CheckIcon } from "@radix-ui/react-icons";
+import { Command, CommandEmpty, CommandGroup, CommandInput, CommandItem } from "@/components/ui/command";
+import { useState } from "react";
+import { Filter, getFilters } from "@/lib/api";
+import { Dialog, DialogContent, DialogDescription, DialogFooter, DialogHeader, DialogTitle } from "../ui/dialog";
+import { Input } from "@/components/ui/input";
+
+interface AdminFilterProps {
+  filterTypeSingular: string;
+  filterTypePlural: string;
+  filterTable: FirebaseTablesEnum;
+  memberId: string;
+  filters: { name: string; id: string; status: string }[];
+  setFilters: (filter: { name: string; id: string; status: string }[]) => void;
+  emoji: string;
+  suggestedFilter: string;
+  setSuggestedFilter: (suggestedFilter: string) => void;
+}
+
+export default function AdminFilter({
+  filterTypeSingular,
```

Comment:
> Do you feel three props for labeling is a bit noisy? Is the emoji necessary?

Like you mentioned in the [other feedback](https://github.com/hawaiians/hawaiiansintech/pull/25#discussion_r1424482210), the heading isn't actually even necessary. I had it to distinguish between focuses and industry since the views would be the same. I'll remove it

> ...what about passing in an array...

yep that makes sense. I was thinking about passing in an object since the parameters were getting pretty big but I'm always a little hesitant about passing objects. The one you suggested is pretty straight forward especially because they're just strings and pretty related to eachother.

Reply:
> labels: [singular: string, plural: string]

oh wait, why exactly an array? Why not an object?: `labels: {singular: string, plural: string}`

Then we call with `labels.singular` rather than `labels[0]`